credentials/java/com/google/auth/ApiKeyCredentials.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2024, Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.auth;
+
+import com.google.api.client.util.*;
+import java.io.IOException;
+import java.net.URI;
+import java.util.*;
+
+/**
+ * Credentials class for calling Google APIs using an API key.
+ *
+ * <p>Uses an API key directly in the request metadata to provide authorization.
+ *
+ * <pre><code>
+ * Credentials credentials = ApiKeyCredentials.create("your api key");
+ * </code></pre>
+ */
+public class ApiKeyCredentials extends Credentials {
```

Comment:
I suppose this class is in `google-auth-library-credentials` module unlike other credential classes we have in `google-auth-library-oauth2-http` module because it is not an OAuth2 credential?
Do you mind adding some context or links to the description section so that I can understand a bit more on this and for future tracking purposes? I find context and links to docs very helpful when tracking back changes in the past.

Reply:
added note in description, let me know if you think it need any additional information